### ISSUE TITLE
slack.com integration

### ISSUE THREAD

**Issue by ashumz** [waffleio/waffle.io#806]

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/667627959)

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/690527612)

**Comment by boblannon** [comment]
+1

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/692718468)

**Comment by woisme** [comment]
+1

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/698587820)

**Comment by garysheng** [comment]
+1

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/ashley@waffle.io/conversations/469578740)

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/707706711)

**Comment by DerrickGrant** [comment]
+1

**Comment by derchrisuk** [comment]
+1

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/709737100)

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/712664158)

**Comment by phae** [comment]
:+1:

**Comment by agibson22** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/713209209)

**Comment by flavioribeiro** [comment]
+1

**Comment by asjer** [comment]
+1

**Comment by ehlersd** [comment]
+1

**Comment by goddamnhippie** [comment]
+1

**Comment by phobologic** [comment]
:+1:

**Comment by jamestoyer** [comment]
:+1:

**Comment by jamescrowley** [comment]
+1

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/721776211)

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/731516716)

**Comment by diophore** [comment]
We already have GitHub integrated with Slack. I would never want two notifications for the same event. I would only want notifications for events specific to waffle that don't cause an identical notification from GitHub.

**Comment by marcellodesales** [comment]
+1 

Until when I have to wait for this integration? I'd like to have the status of the team and track the events that happens on the board such as state transition. Everything that's NOT part of GitHub integration.

**Comment by jean** [comment]
This +1'ing is pretty nasty :-( clutter and noise.

**Comment by ashumz** [comment]
@diophore That's a good point -- what notifications would you want to see from Waffle that you don't already see from GitHub? 

For those that would want a Waffle integration with Slack, would you want it to replace the integration you're running with GitHub or have notifications from both? Which types of notifications would be most useful to you and/or your team? 

Thanks! 

Ashley

**Comment by marcellodesales** [comment]
Hi @ashumz,

Imagine notifications being the events in a regular board:

* New Issues and Assignment changes
* Cards transitions from one step to the next
* When a card was closed or not
* When a card is marked as READY or BLOCKED

Do you have an estimate of when would this integration come in? 

thanks!

**Comment by diophore** [comment]
GitHub already notifies Slack when pull requests and issues are created or closed. I think the best solution for Waffle is to allow us to configure exactly which notifications we would like to receive from Waffle. Using @marcellodesales's list as a starting point, I would allow us to choose from these options:
* New issue created (I would disable this, preferring GitHub to send this one)
* New comment on issue (same, I would disable)
* Issue assigned
* Issue moved to another column (except closed)
* Issue marked blocked
* Issue closed (I would disable)

Lastly, and this is related to a separate feature I would like to see added, I would like to receive a notification whenever we exceed the Work in Progress Limit of any column.

**Comment by c4m3lo** [comment]
+1

**Comment by Snugug** [comment]
:+1: I agree with the lists put forth above, plus I'd like to see a notification for when size changes.

**Comment by Lucien** [comment]
+1

**Comment by mattfletcher** [comment]
+1

**Comment by erichhuang** [comment]
+1 on @marcellodesales and @diophore 's suggestions

**Comment by JakeCooper** [comment]
+1

**Comment by bitHero** [comment]
+1

**Comment by bcm** [comment]
+1

**Comment by ajkavanagh** [comment]
+1 for integration, but also for filtering out notifications that github would already send as I also have an integration between github and slack.

**Comment by johnniehare** [comment]
+1

**Comment by nickbouton** [comment]
+1

**Comment by pjm17971** [comment]
+1

**Comment by danielBingham** [comment]
+1

**Comment by qualityshepherd** [comment]
+1

**Comment by babakgh** [comment]
How many +1 they need?!

**Comment by pathikrit** [comment]
@babakgh : 1024

**Comment by robcaw** [comment]
+1

**Comment by devinus** [comment]
:+1:

**Comment by CarlRosell** [comment]
:+1:

**Comment by nditada** [comment]
+1

**Comment by Leezlee** [comment]
+1

**Comment by zachthieme** [comment]
+1

**Comment by underc0de** [comment]
+1

**Comment by statik** [comment]
:thumbsup:

**Comment by rockerhieu** [comment]
:+1:

**Comment by adrianolisboa** [comment]
:+1:

**Comment by aaronmallen** [comment]
:+1:

**Comment by cpostma-rally** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/937154447)

**Comment by mbulkeley** [comment]
+1

**Comment by NanoDami** [comment]
+1

**Comment by samervin** [comment]
+1

**Comment by arthurthefourth** [comment]
+1

**Comment by burtonjc** [comment]
:+1: [from intercom](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/949120797)

**Comment by cpostma-rally** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/949063343)

**Comment by cpostma-rally** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/953962757)

**Comment by Anahkiasen** [comment]
:+1:

**Comment by cpostma-rally** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/964470086)

**Comment by cpostma-rally** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/965443468)

**Comment by luckychild305** [comment]
+1

**Comment by bbirch** [comment]
+1

**Comment by utgarda** [comment]
+1

**Comment by colbygk** [comment]
+1

**Comment by spheromak** [comment]
:100:

**Comment by JonathanPorta** [comment]
+1

**Comment by fujin** [comment]
+++ :100:

**Comment by timani** [comment]
+1

**Comment by dpedoneze** [comment]
+1

I think the _most-used_ notification would be when we change an issue among the columns, GitHub doesn't has this kind of notification (label-change) :smile:

**Comment by Cram974** [comment]
+1

**Comment by charlescurrier** [comment]
+1

**Comment by luislew** [comment]
+1

**Comment by scarolan** [comment]
+1

**Comment by ggdaltoso** [comment]
+1

**Comment by jmeridth** [comment]
:+1:

**Comment by jjman505** [comment]
:+1:

**Comment by brucellino** [comment]
Wow... could this be the most popular request EVAR!? :+1:

**Comment by mwasn** [comment]
+1

**Comment by fizruk** [comment]
+1

**Comment by paradoxxjm** [comment]
+1

**Comment by alleyhector** [comment]
:+1:

**Comment by idris** [comment]
:+1:

**Comment by weaversam8** [comment]
+1

**Comment by karthikrengarajan** [comment]
+1

**Comment by lukas-zech-software** [comment]
:+1: 

Periodically sending a message like "There are open PullRequests available" would be nice to remind the team, that there is still work to be done!

**Comment by alanguerin** [comment]
:thumbsup:

**Comment by devoxel** [comment]
:+1:

**Comment by Cyb3rWarri0r8** [comment]
:+1:

**Comment by ashumz** [comment]
[internal request](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/ashley@waffle.io/conversations/1319485917)

We already use the Github Integration for Slack which tells us, that when there are new pull requests created.

Our problem is, that our developers are working at their own tasks and may see the pull request notification from Github, but they don't have time to work on in right now. But they tend to forget about them later and do not visit the waffle board frequently

What we imagined from a Waffle Slack integration would be something that periodically reminds us, that there are still open pull requests waiting to be reviewed.

Every our or so, Waffle could send a message "There are still 2 pull request waiting on Board 'Waffle-1' "
Or after a distinct threshold of pull requests in the Back Log lane the message could be sent with red background and more often, so that the developers are urged to work on the pull requests.
After a second threshold is reached, or a pull request has not been process after a maximum time/age there could be a message to the lead developer, who can then pick a developer by hand and order him to do the PR."

**Comment by paradoxxjm** [comment]
@ashumz that seems like a great way of thinking about the integration. In particular for our team, we sometimes suffer from "zombie" pull requests or issues in our "review" board that have been sitting there for a while. It would be great if the Waffle integration allowed one to flag a particular board to publish reminders as you are suggesting over a defined period of time.

**Comment by ytvinay** [comment]
+1

**Comment by sguimont** [comment]
+1

**Comment by raphydaphy** [comment]
:+1:

**Comment by sabbyanandan** [comment]
:thumbsup:

**Comment by therealbill** [comment]
:+1:

**Comment by appelsiini** [comment]
:+1:

**Comment by galenandrew** [comment]
:+1:

**Comment by htsh-tsyk** [comment]
+1

**Comment by kankaristo** [comment]
+1, but for us this would replace the GitHub integration.

We're on Slack's free plan, and the only reason we would switch to the paid plan (~400 $ per year for us) is if we hit the integration limit (we're at 9/10 right now). We're cheapskates.

**Comment by zboralski** [comment]
+1

**Comment by dmehra** [comment]
:+1:

**Comment by localredhead** [comment]
:100:

**Comment by localredhead** [comment]
It would be great to be able to tag or pin a conversation to an open issue somehow.

**Comment by ginfuru** [comment]
:+1:

**Comment by Yolgie** [comment]
+1

**Comment by janaka** [comment]
+1

**Comment by schlos** [comment]
+1

I know about slacks  integration for GitHub, but then I get all commits and pull requests in Slack, while I'd like to get only issues changes (like what Waffle is doing, status changes etc)

**Comment by braddock-g** [comment]
:+1:

**Comment by LoLei** [comment]
:+1:

**Comment by kyokutyo** [comment]
+1

**Comment by agrizzlyadams** [comment]
:+1:

**Comment by shepsii** [comment]
+1

**Comment by kiwibonga** [comment]
+1

Slack's built-in Github integration is incomplete (many hooks are just not implemented), so this could be a killer feature for Slack + Github users.

This has been accumulating +1s for two years, let's get the ball rolling!

**Comment by freeranger** [comment]
+1

**Comment by mskutin** [comment]
:+1:

**Comment by myarotskaya** [comment]
:+1:

**Comment by ebashmakov** [comment]
:+1:

**Comment by stevecox** [comment]
:+1: 
The power of the Slack integration comes from centralizing project notifications, especially when using multi-repo.  This way I only have to setup the waffle.io integration and not a webhook for every repo in GH.

**Comment by cee-dub** [comment]
+1

**Comment by agrizzlyadams** [comment]
+1

**Comment by vjpr** [comment]
+1

**Comment by jtewright** [comment]
+1

**Comment by jaimerump** [comment]
+1

**Comment by teebash** [comment]
:+1:

**Comment by wavded** [comment]
+1 -- come-on @github let's get that voting thing implemented :)

**Comment by kilianc** [comment]
+1 this is so funny.

**Comment by corysimmons** [comment]
![](http://i.imgur.com/z4POxDC.gif)

**Comment by niksmac** [comment]
+1

**Comment by corysimmons** [comment]
Please add +1's as reactions from here on out.

![image](https://cloud.githubusercontent.com/assets/224636/13694919/edab7938-e727-11e5-964a-7c3cec1b518e.png)

**Comment by hwinkel** [comment]
:+1:

**Comment by brentrobbins** [comment]
+1

**Comment by kilodalton** [comment]
+1

**Comment by Yorkshireman** [comment]
+1

**Comment by zach-wildfig** [comment]
+1

**Comment by jdegger** [comment]
+1

**Comment by kirly-af** [comment]
+1

**Comment by kirillseva** [comment]
:+1:

**Comment by jarradh** [comment]
+1

**Comment by hakan42** [comment]
+1

**Comment by EliseTemple** [comment]
👍

**Comment by CathyMacars** [comment]
+1

**Comment by EliseTemple** [comment]
It would also be great to get messages when things move form one column to another.  adding github to slack allows you to see any comments, and new commits, but when tags or added or changed, it doesn't get sent to slack.

**Comment by hspinks** [comment]
+1

**Comment by nadeemyasin61** [comment]
+1

**Comment by kave** [comment]
👍

**Comment by scardino** [comment]
+1

**Comment by dbuarque** [comment]
+1

**Comment by mikeric** [comment]
+1

**Comment by timelf123** [comment]
![http://i.imgur.com/fHySyO1.png](http://i.imgur.com/fHySyO1.png)

**Comment by niksmac** [comment]
+1

**Comment by lieldulev** [comment]
:+1

**Comment by joehanna** [comment]
+1

**Comment by Copywright** [comment]
+1

**Comment by philhudson91** [comment]
+69

**Comment by jdegger** [comment]
@ashumz it has been since... 2014? There are 162 participants in this conversation and this is the second most requested feature on here (after Gitlab).

<img width="970" alt="screen shot 2016-09-09 at 12 30 44" src="https://cloud.githubusercontent.com/assets/14085897/18384148/4ef30620-7689-11e6-9af6-b7e72c743e2c.png">

Are guys still actively developing or should we just give up hope features like this are going to be built? I mean integrating something like Slack is honestly a day of work.. if you work slow.

**Comment by steveacalabro** [comment]
+1 again and what the guy above me said is true

**Comment by mmmmmrob** [comment]
👍

**Comment by LuisBurgos** [comment]
+1

**Comment by steefmin** [comment]
+1

**Comment by daveboling** [comment]
+1

**Comment by freeall** [comment]
Is there any progress on this?

**Comment by kellihsf** [comment]
hi @freeall,

Yep! We're currently working on this now, we hope to have it out very soon. 😀

**Comment by kellihsf** [comment]
Hi all,

We've so excited to announce we've just released our Slack integration for Waffle! Check out our [blog](http://blog.waffle.io/slack_integration) to learn more!

**Comment by brucellino** [comment]
But.... I've already moved all my repos to ZenHub ... but.... I also really like waffles .... but.... the zenhub hooks and columns are set up just right.... but.... waffle came first and provides an actual url instead of a browser plugin that I need to ask my collaborators to use .... but... zehub... but... waffle.... 

![xlr2via2mvfmk](https://cloud.githubusercontent.com/assets/2115428/23021015/162fde64-f44a-11e6-9f23-1c8384935430.gif)

OK, I'll shut up, y'all make a good product and I'm using  it for free, so :clap: :clap: thanks for this, it's greatly appreciated :tada:
